### PR TITLE
feat(notifier): adds funds withdraw as provider

### PR DESCRIPTION
**Pull request by jurajpiar**
Resolves https://rsklabs.atlassian.net/browse/RMKT-626

Depends on https://github.com/rsksmart/rif-marketplace-cache/pull/661

### CODE REVIEW

File: src/contracts/notifier/Notifier.ts [hunk 1]
```diff
@@ -57,6 +60,28 @@ class NotifierContract extends ContractWithTokens {
...
   }
 
+  public async withdrawFunds({
+    hash, token: { tokenAddress }, amount, from,
+  }: {
+    hash: string
+    token: BaseToken
+    amount: Big
+    from: string
+  }, txOptions: TxOptions = {}): Promise<TransactionReceipt> {
```
Comment (0xartem) [inline src/contracts/notifier/Notifier.ts]: let's make it a `type`

**Review by 0xartem — APPROVED** [review]
overall looks good

File: src/components/templates/ProgressOverlay.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ const ProgressOverlay: FC<Props> = ({
                 {
                   buttons.map((btn) => (
                     <GridItem
-                      key={btn.key || (btn as unknown as HTMLElement).innerText}
```
Comment (itofarina) [inline src/components/templates/ProgressOverlay.tsx:48]: Why would we stop using `btn.key` prop when available here?

**Review by itofarina — COMMENTED** [review]
Apart from that small comment LGTM

File: src/components/templates/ProgressOverlay.tsx (hunk 2 above)
Comment (jurajpiar) [reply src/components/templates/ProgressOverlay.tsx:48]: ah yea..tx

**Pull request merged**